airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java

```diff
@@ -338,70 +328,58 @@ public ManualSyncSubmissionResult startNewManualSync(final UUID connectionId) {
 
     final long jobId = connectionManagerWorkflow.getJobInformation().getJobId();
 
-    return new ManualSyncSubmissionResult(
+    return new ManualOperationResult(
         Optional.empty(),
         Optional.of(jobId));
   }
 
-  @Value
-  public class NewCancellationSubmissionResult {
-
-    final Optional<String> failingReason;
-    final Optional<Long> jobId;
-
-  }
-
-  public ManualSyncSubmissionResult startNewCancelation(final UUID connectionId) {
-    log.info("Manual sync request");
-
-    final boolean workflowReachable = isWorkflowReachable(getConnectionManagerName(connectionId));
+  public ManualOperationResult startNewCancellation(final UUID connectionId) {
+    log.info("Manual cancellation request");
 
-    if (!workflowReachable) {
-      log.error("Can't cancel a non running workflow");
-      return new ManualSyncSubmissionResult(
-          Optional.of("No scheduler workflow is reachable for: " + connectionId),
+    final ConnectionManagerWorkflow connectionManagerWorkflow;
+    try {
+      connectionManagerWorkflow = repairAndRetrieveConnectionManagerWorkflow(connectionId);
+    } catch (final DeletedWorkflowException e) {
+      log.error("Can't cancel a deleted workflow");
+      return new ManualOperationResult(
+          Optional.of(e.getMessage()),
           Optional.empty());
     }
```

Comment:
this block of code is repeated a bunch of times in this class (except for the contents of the log message). can we DRY it?

Reply:
I think this is a hard piece of logic to DRY, because this logic means that _this_ method should return a ManualOperationResult in the case of a DeletedWorkflowException. So if I try to move this logic into another method, that method will need to somehow indicate to this method that it should return a ManualOperationResult in that case, or return a ConnectionManagerWorkflow in the normal case, and this method will need to handle both cases with a conditional. So I think either way, this will be somewhat ugly if we try to DRY it.